hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -874,21 +887,30 @@ private static Object rewritePrimaryTypeWithDiffSchemaType(Object oldValue, Sche
     switch (newSchema.getType()) {
       case NULL:
       case BOOLEAN:
+        if (oldSchema.getType() == Schema.Type.STRING) {
```

Comment:
Appreciate your approach of extending existing functionality, but i think in that case we actually shouldn't be doing that -- conversions we're doing here aren't "canonical" (there's no permitted conversion from string to int, for ex in Spark) and as such, we'd rather keep this conversion limited in scope to be applicable only to partition values where it's both inevitable and is actually sensible (since we know that we're not trying to convert some random string to int, but the int that was previously converted to a string)

Reply:
Also i'd suggest you to check out `Spark3ParsePartitionUtil`

Reply:
Hi @alexeykudinkin, thanks for your suggestion very much. 
What do you think about wrapping logFiles as PartitionedFiles, just like basefiles?